virtcontainers/kata_agent.go

```diff
@@ -461,6 +464,19 @@ func (k *kataAgent) startSandbox(sandbox *Sandbox) error {
 		hostname = hostname[:maxHostnameLen]
 	}
 
+	//check grpc server is serving
+	if err = k.connect(); err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), grpcCheckTimeout)
+	defer cancel()
+	if _, err = k.client.Check(ctx, &grpc.CheckRequest{}); err != nil {
```

Comment:
This looks good, but I was thinking that in order to avoid duplication of code, the check with timeout should be available through the agent interface itself. Thus, you could enhance the agent interface regarding `check()` function, and then rely on this new version to call it from here.

Reply:
You can simply use k.check() here, and it handles grpc connect/disconnect properly.

Reply:
@bergwolf  k.check doesn‘t have `timeout`. Need to enhance it.

Reply:
@jshachm Take a look at `installReqFunc()`. Check request has builtin a 5 second timeout, same as you defined in this PR.

Reply:
@bergwolf  sry for miss it~~  Yep,  with the builtin timeout, `k.check` will make a difference.

Reply:
@bergwolf Thanks, it's a great suggestion,  it makes code more clean and simple